### PR TITLE
feat(csp): add reportOnly option

### PR DESCRIPTION
**Pull request by TheAlexLichter**
This PR adds a `reportOnly` option to the `render.csp` object that's enabled when in `dev mode` (more specific: when `options.debug` is true) by default.

It will change the header from the usual `Content-Security-Policy` header to the `Content-Security-Policy-Report-Only` header. When this header is set, the CSP isn't fully enabled but provides information about violations of the CSP and how to fix them.

### CODE REVIEW

**Comment by manniL** [comment]
Tests are failing, but not bc of the code change :thinking: 

```
FAIL test/fixtures/with-config/with-config.test.js (81.903s)
  ● Build with-config
    Timeout - Async callback was not invoked within the 60000ms timeout specified by jest.setTimeout.
      2 | 
      3 | export const buildFixture = function buildFixture(fixture) {
    > 4 |   test(`Build ${fixture}`, async () => {
        |   ^
      5 |     const config = loadFixture(fixture, {
      6 |       test: true,
      7 |       build: {
      at Spec (node_modules/jest-jasmine2/build/jasmine/Spec.js:85:20)
      at buildFixture (test/utils/build.js:4:3)
      at Object.<anonymous> (test/fixtures/with-config/with-config.test.js:3:1)
```

**Review by clarkdo — COMMENTED** [review]
~~How about moving static and dynamic (reportOnly) `cspDefaults`to `lib/common/nuxt.confg.js` and `lib/common/options.js` ?~~
Ignore above, I forgot I have moved them.

File: lib/core/middleware/nuxt.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ export default async function nuxtMiddleware(req, res, next) {
 
     if (this.options.render.csp) {
       const { allowedSources, policies } = this.options.render.csp
-      let cspStr = `script-src 'self'${this.options.dev ? " 'unsafe-eval'" : ''} ${(cspScriptSrcHashes).join(' ')}`
+      let cspStr = `script-src 'self'${this.options.dev ? ` 'unsafe-eval'` : ''} ${(cspScriptSrcHashes).join(' ')}`
```
Comment (clarkdo) [inline lib/core/middleware/nuxt.js]: Is this change necessary ?

**Review by clarkdo — COMMENTED** [review]
LGTM

File: lib/core/middleware/nuxt.js (hunk 1 above)
Comment (manniL) [reply lib/core/middleware/nuxt.js]: Not necessary, did this because my editor settings replace `"'...'"` with `'\'...\''`. Should I revert it?

Comment (clarkdo) [reply lib/core/middleware/nuxt.js]: I think we can:

```js
let cspStr = `script-src 'self'${this.options.dev ? ' unsafe-eval' : ''} ${(cspScriptSrcHashes).join(' ')}`
```

Comment (manniL) [reply lib/core/middleware/nuxt.js]: The "double quotes" are necessary.

The final must should look like: `script-src 'self' 'unsafe-eval' ...`

Comment (clarkdo) [reply lib/core/middleware/nuxt.js]: Sorry, my mistake, I didn't notice that.

**Comment by Atinux** [comment]
Sounds good to me, do you mind resolving the actual conflicts @manniL ?

**Comment by manniL** [comment]
@Atinux I've resolved the conflicts and adapted the tests a little. Should be fine now :)
